internal/common/objects/BikerAgent.go

```diff
@@ -0,0 +1,59 @@
+package baseextendedagent
+
+/*
+	BikerAgent is the implentation of IBikerAgent. All extended biker agents must be composed of a BikerAgent object
+*/
+
+import (
+	baseagent "github.com/MattSScott/basePlatformSOMAS/BaseAgent"
+	"github.com/google/uuid"
+	// "SOMAS2023/internal/common/utils"
+)
+
+type BikerAgent struct {
+	colour Colour
+	forces Forces
+	position [2]float64
+}
+
+// generates a new BikerAgent instance with a randomised ID
+NewBikerAgent [T IBikerAgent [ T ]]() ∗ BikerAgent [ T ] {
+	return &BaseAgent [ T] {
+		id uuid.ID
+	}
+}
+
+// SetColour sets the color of the BikerAgent.
+func (ba *BikerAgent) SetColour(lootBoxColour Colour) {
+	ba.colour = lootBoxColour
+}
+
+// GetColour returns the color of the BikerAgent.
+func (ba *BikerAgent) GetColour() Colour {
+	return ba.colour
+}
+
+// GetForces returns the current forces of the BikerAgent.
+func (ba *BikerAgent) GetForces() Forces {
+	return ba.forces
+}
+
+// UpdateForces updates the forces based on some decision-making process.
+func (ba *BikerAgent) UpdateForces() {
```

Comment:
this will probably need some arguments, so that you can actually update the forces. 

In fact, I don't think that the forces should be a member of the struct - surely you'll just return it from 'GetForces', say?